script.js

```diff
@@ -0,0 +1,155 @@
+const apiKey = 'DBELFYET2HB66AN6SZC56BC6M';
+const form = document.getElementById('search-form');
+const input = document.getElementById('location-input');
+const weatherDisplay = document.getElementById('weather-display');
+const forecastDisplay = document.getElementById('forecast-display');
+const toggleButton = document.getElementById('toggle-units');
+const loadingIndicator = document.getElementById('loading');
+const errorMessage = document.getElementById('error-message');
+
+let currentUnit = 'metric'; // metric = Celsius, us = Fahrenheit
+let lastLocation = '';
+
+form.addEventListener('submit', (e) => {
+    e.preventDefault();
+    const location = input.value.trim();
+    if (location) {
```

Comment:
I feel you can add input validation and feedback to improve user experience... Handle empty or invalid input more clearly